core/sector_storage/impl/sector_storage_impl.cpp

```diff
@@ -0,0 +1,217 @@
+/**
+ * Copyright Soramitsu Co., Ltd. All Rights Reserved.
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+#include "sector_storage/impl/sector_storage_impl.hpp"
+#include <fcntl.h>
+#include "sector_storage/sector_storage_error.hpp"
+
+namespace fc::sector_storage {
+
+  using fc::primitives::piece::PaddedPieceSize;
+  using fc::primitives::sector_file::SectorFileTypes;
+  using proofs = fc::proofs::Proofs;
+  using fc::primitives::sector::getSectorSize;
+
+  SectorStorageImpl::SectorStorageImpl(const std::string &root_path,
+                                       RegisteredProof post_proof,
+                                       RegisteredProof seal_proof)
+      : root_(root_path),
+        seal_proof_type_(seal_proof),
+        post_proof_type_(post_proof) {
+    size_ = 0;
+    auto s1 = getSectorSize(seal_proof);
+    if (s1.has_value()) {
+      size_ = s1.value();
+    }
+  }
+
+  outcome::result<SectorPaths> SectorStorageImpl::acquireSector(
+      SectorId id,
+      const SectorFileType &existing,
+      const SectorFileType &allocate,
+      bool sealing) {
+    auto cache_path =
```

Comment:
Repetitive. Maybe we can create directory only of requested type, and others will be created lazily when requested?

Reply:
Fixed